tests/Doctrine/Tests/Common/DataFixtures/Purger/ORMPurgerForeignKeyCheckTest.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\Common\DataFixtures;
+
+use Doctrine\Common\DataFixtures\Purger\ORMPurger;
+use Doctrine\Common\EventManager;
+use Doctrine\DBAL\Configuration;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Driver\Middleware\AbstractDriverMiddleware;
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Platforms\MySQLPlatform;
+use Doctrine\DBAL\Platforms\SqlitePlatform;
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\Mapping\ClassMetadata;
+use Doctrine\ORM\ORMSetup;
+use Doctrine\Persistence\Mapping\Driver\MappingDriver;
+use Doctrine\Tests\Common\DataFixtures\TestEntity\Link;
+
+class ORMPurgerForeignKeyCheckTest extends BaseTest
+{
+    public const TEST_CLASS_NAME = Link::class;
+    public const TEST_TABLE_NAME = 'link';
+
+    /** @return MappingDriver */
+    protected function getMockMetadataDriver()
+    {
+        $metadataDriver = $this->createMock(MappingDriver::class);
+        $metadataDriver->method('getAllClassNames')->willReturn([self::TEST_CLASS_NAME]);
+        $metadataDriver->method('loadMetadataForClass')
+            ->willReturnCallback(static function (string $className, ClassMetadata $metadata) {
+                if ($className !== self::TEST_CLASS_NAME) {
+                    return;
+                }
+
+                $metadata->setPrimaryTable(['name' => self::TEST_TABLE_NAME]);
+                $metadata->setIdentifier(['id']);
+            });
+        $metadataDriver->method('isTransient')->willReturn(false);
+
+        return $metadataDriver;
+    }
+
+    /** @return Connection */
+    protected function getMockConnectionForPlatform(AbstractPlatform $platform)
```

Comment:
```suggestion
    /** @return Connection&MockObject */
    protected function getMockConnectionForPlatform(AbstractPlatform $platform): Connection
```

Reply:
Yes! Let me figure out how i can add the change to the previous function via this UI.